Src/Base/AMReX_AsyncOut.cpp

```diff
@@ -41,12 +41,20 @@ void Initialize ()
     if (s_asyncout and s_noutfiles < nprocs)
     {
 #ifdef AMREX_MPI_THREAD_MULTIPLE
+        int provided = -1;
+        MPI_Query_thread(&provided);
+        if (provided < MPI_THREAD_MULTIPLE)
+            amrex::Abort("AsyncOut with " + std::to_string(s_noutfiles) + " and "
+                         +std::to_string(nprocs) + " processes requires "
+                         +"MPI_THREAD_MULTIPLE at runtime");
+
         int myproc = ParallelDescriptor::MyProc();
         s_info = GetWriteInfo(myproc);
         MPI_Comm_split(ParallelDescriptor::Communicator(), s_info.ifile, myproc, &s_comm);
 #else
```

Comment:
I would argue we could also remove the compile-time part here at all, if you like.

Reply:
Yes.